bazel-watcher is broken on Darwin

**Describe the bug**

`bazel-watcher` is not building on Darwin.

**To Reproduce**
Steps to reproduce the behavior:
1. `nix-build -A bazel-watcher`

**Expected behavior**
Bazel watcher correctly built.

**Additional context**

`bazel-watcher` was broken by 4dad0ae560708fee5d081c3f02bf94e4c6f4dfce. cc @guibou @Profpatsch @abbradar.

<details>
<summary>Build log</summary>

```console
unpacking sources
unpacking source archive /nix/store/6waa28r78jwaarcbwh14vd2i8lzgrnhl-source
source root is source
patching sources
applying patch /nix/store/jh9xn6gchwxn479l63aqrhgv3cr43n4n-use-go-in-path.patch
patching file WORKSPACE
applying patch /nix/store/c2nssk695bl2scdfvb0idrna2w12298k-686130f50cea274f7453f6abc8c5249654047462.patch
patching file WORKSPACE
applying patch /nix/store/fn6v9qklx9pbz974868pws1dhjkx7qfi-18bdb44832ccc533e0ab3923ef80060eeb24582d.patch
patching file WORKSPACE
configuring
no configure script, doing nothing
building
patching script interpreter paths in .
./release/release.sh: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/d9wxsyxsj3ssjcbywwm35254d14yv87a-bash-4.4-p23/bin/bash"
Extracting Bazel installation...
Starting local Bazel server and connecting to it...
Loading:
Loading: 0 packages loaded
Analyzing: target //ibazel:ibazel (1 packages loaded, 0 targets configured)
Analyzing: target //ibazel:ibazel (50 packages loaded, 6806 targets configured)
Analyzing: target //ibazel:ibazel (53 packages loaded, 7648 targets configured)
INFO: Analyzed target //ibazel:ibazel (53 packages loaded, 7784 targets configured).
INFO: Found 1 target...
[0 / 53] [Prepa] Writing file external/go_sdk/packages.txt [for host] ... (3 actions, 0 running)
INFO: From Executing genrule @net_zlib//:copy_public_headers [for host]:
/nix/store/9a77gahrjfv86qj3d396j1jm6ypvyhac-bash/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US.ISO-8859-1): No such file or directory
INFO: From Executing genrule @com_google_protobuf//:_internal_wkt_protos_genrule:
/nix/store/9a77gahrjfv86qj3d396j1jm6ypvyhac-bash/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US.ISO-8859-1): No such file or directory
[10 / 278] [Scann] Compiling external/net_zlib/uncompr.c [for host] ... (2 actions, 0 running)
ERROR: /private/var/folders/ck/ht80_l5d4c9_tnwmd5lqsxj40000gn/T/nix-build-bazel-watcher-0.10.3.drv-0/output/external/com_google_protobuf/BUILD:388:1: C++ compilation of rule '@com_google_protobuf//:protoc' failed (Exit 1) cc_wrapper.sh failed: error executing command external/local_config_cc/cc_wrapper.sh -U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -fcolor-diagnostics -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG ... (remaining 38 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
/nix/store/9a77gahrjfv86qj3d396j1jm6ypvyhac-bash/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US.ISO-8859-1): No such file or directory
/nix/store/d9wxsyxsj3ssjcbywwm35254d14yv87a-bash-4.4-p23/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US.ISO-8859-1): No such file or directory
In file included from external/com_google_protobuf/src/google/protobuf/compiler/main.cc:33:
external/com_google_protobuf/src/google/protobuf/compiler/cpp/cpp_generator.h:40:10: fatal error: 'string' file not found
#include <string>
         ^~~~~~~~
1 error generated.
Target //ibazel:ibazel failed to build
Use --verbose_failures to see the command lines of failed build steps.
ERROR: /private/var/folders/ck/ht80_l5d4c9_tnwmd5lqsxj40000gn/T/nix-build-bazel-watcher-0.10.3.drv-0/source/ibazel/BUILD:17:1 C++ compilation of rule '@com_google_protobuf//:protoc' failed (Exit 1) cc_wrapper.sh failed: error executing command external/local_config_cc/cc_wrapper.sh -U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -fcolor-diagnostics -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG ... (remaining 38 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
INFO: Elapsed time: 7.600s, Critical Path: 0.24s
INFO: 2 processes: 2 processwrapper-sandbox.
FAILED: Build did NOT complete successfully
FAILED: Build did NOT complete successfully
builder for '/nix/store/cayv9fpjg09kr5i9mmriyd7nhdscn0dq-bazel-watcher-0.10.3.drv' failed with exit code 1
error: build of '/nix/store/cayv9fpjg09kr5i9mmriyd7nhdscn0dq-bazel-watcher-0.10.3.drv' failed
```
</details>

**Metadata**
Please run `nix run nixpkgs.nix-info -c nix-info -m` and paste the result.

 - system: `"x86_64-darwin"`
 - host os: `Darwin 18.6.0, macOS 10.14.5`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.2.2`
 - channels(yl): `"darwin, nixpkgs-19.09pre185256.5b8b5a694c7"`
 - nixpkgs: `/run/current-system/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: bazel-watcher
```

--- a/pkgs/build-support/build-bazel-package/default.nix
+++ b/pkgs/build-support/build-bazel-package/default.nix
@@ -119,7 +119,7 @@ in stdenv.mkDerivation (fBuildAttrs // {
     # wrappers picking them up, pass them in explicitly via `--copt`, `--linkopt`
     # and related flags.
     #
-    '' + (lib.optionalString stdenv.isDarwin ''
+    '' + lib.optionalString stdenv.isDarwin ''
     copts=()
     host_copts=()
     for flag in $NIX_CFLAGS_COMPILE; do
@@ -136,7 +136,7 @@ in stdenv.mkDerivation (fBuildAttrs // {
       linkopts+=( "--linkopt=$flag" )
       host_linkopts+=( "--host_linkopt=$flag" )
     done
-    '') + ''
+    '' + ''
 
     BAZEL_USE_CPP_ONLY_TOOLCHAIN=1 \
     USER=homeless-shelter \
@@ -145,12 +145,12 @@ in stdenv.mkDerivation (fBuildAttrs // {
       --output_user_root="$bazelUserRoot" \
       build \
       -j $NIX_BUILD_CORES \
-      '' + (lib.optionalString stdenv.isDarwin ''
+      '' + lib.optionalString stdenv.isDarwin ''
       "''${copts[@]}" \
       "''${host_copts[@]}" \
       "''${linkopts[@]}" \
       "''${host_linkopts[@]}" \
-      '') + ''
+      '' + ''
       $bazelFlags \
       $bazelTarget
 

--- a/pkgs/build-support/build-bazel-package/default.nix
+++ b/pkgs/build-support/build-bazel-package/default.nix
@@ -109,6 +109,29 @@ in stdenv.mkDerivation (fBuildAttrs // {
   buildPhase = fBuildAttrs.buildPhase or ''
     runHook preBuild
 
+    # Bazel sandboxes the execution of the tools it invokes, so even though we are
+    # calling the correct nix wrappers, the values of the environment variables
+    # the wrappers are expecting will not be set. So instead of relying on the
+    # wrappers picking them up, pass them in explicitly via `--copt`, `--linkopt`
+    # and related flags.
+    #
+    copts=()
+    host_copts=()
+    for flag in $NIX_CFLAGS_COMPILE; do
+      copts+=( "--copt=$flag" )
+      host_copts+=( "--host_copt=$flag" )
+    done
+    for flag in $NIX_CXXSTDLIB_COMPILE; do
+      copts+=( "--copt=$flag" )
+      host_copts+=( "--host_copt=$flag" )
+    done
+    linkopts=()
+    host_linkopts=()
+    for flag in $NIX_LD_FLAGS; do
+      linkopts+=( "--linkopt=$flag" )
+      host_linkopts+=( "--host_linkopt=$flag" )
+    done
+
     BAZEL_USE_CPP_ONLY_TOOLCHAIN=1 \
     USER=homeless-shelter \
     bazel \
@@ -116,6 +139,10 @@ in stdenv.mkDerivation (fBuildAttrs // {
       --output_user_root="$bazelUserRoot" \
       build \
       -j $NIX_BUILD_CORES \
+      "''${copts[@]}" \
+      "''${host_copts[@]}" \
+      "''${linkopts[@]}" \
+      "''${host_linkopts[@]}" \
       $bazelFlags \
       $bazelTarget
 

--- a/pkgs/build-support/build-bazel-package/default.nix
+++ b/pkgs/build-support/build-bazel-package/default.nix
@@ -113,13 +113,13 @@ in stdenv.mkDerivation (fBuildAttrs // {
   buildPhase = fBuildAttrs.buildPhase or ''
     runHook preBuild
 
+    '' + lib.optionalString stdenv.isDarwin ''
     # Bazel sandboxes the execution of the tools it invokes, so even though we are
     # calling the correct nix wrappers, the values of the environment variables
     # the wrappers are expecting will not be set. So instead of relying on the
     # wrappers picking them up, pass them in explicitly via `--copt`, `--linkopt`
     # and related flags.
     #
-    '' + lib.optionalString stdenv.isDarwin ''
     copts=()
     host_copts=()
     for flag in $NIX_CFLAGS_COMPILE; do

--- a/pkgs/build-support/build-bazel-package/default.nix
+++ b/pkgs/build-support/build-bazel-package/default.nix
@@ -1,4 +1,8 @@
-{ stdenv, bazel, cacert }:
+{ stdenv
+, bazel
+, cacert
+, lib
+}:
 
 args@{ name, bazelFlags ? [], bazelTarget, buildAttrs, fetchAttrs, ... }:
 
@@ -115,6 +119,7 @@ in stdenv.mkDerivation (fBuildAttrs // {
     # wrappers picking them up, pass them in explicitly via `--copt`, `--linkopt`
     # and related flags.
     #
+    '' + (lib.optionalString stdenv.isDarwin ''
     copts=()
     host_copts=()
     for flag in $NIX_CFLAGS_COMPILE; do
@@ -131,6 +136,7 @@ in stdenv.mkDerivation (fBuildAttrs // {
       linkopts+=( "--linkopt=$flag" )
       host_linkopts+=( "--host_linkopt=$flag" )
     done
+    '') + ''
 
     BAZEL_USE_CPP_ONLY_TOOLCHAIN=1 \
     USER=homeless-shelter \
@@ -139,10 +145,12 @@ in stdenv.mkDerivation (fBuildAttrs // {
       --output_user_root="$bazelUserRoot" \
       build \
       -j $NIX_BUILD_CORES \
+      '' + (lib.optionalString stdenv.isDarwin ''
       "''${copts[@]}" \
       "''${host_copts[@]}" \
       "''${linkopts[@]}" \
       "''${host_linkopts[@]}" \
+      '') + ''
       $bazelFlags \
       $bazelTarget
 

